### DISCUSSION TITLE
Cant launch windows uno 5.2 sdk app from vs2022

### DISCUSSION THREAD

**Discussion by cconner100** [unoplatform/uno#16645] [General]
I am in the process of converting a few projects from Microsoft.NET.Sdk to Uno.Sdk.  2 of my 3 projects have converted and can run from within VS2022 preview 7.0 but one can't be launched.  I get the following dialog box when I try to debug

DEP1700: The recipe file "C:\Users\curtisconner\source\repos\ZiiDMSV2\Client\Desktop\ZiiDMSApp\ZiiDMSApp\bin\Debug\net8.0-windows10.0.22621\ZiiDMSApp.build.appxrecipe" does not exist. You may need to build your project.

I have checked that i have deploy selected from the configuration, I have also looked at others who have had this issue and their solutions don't seem to work with this given problem.

[net8.0-windows10.0.22621-Debug-win-arm64.zip](https://github.com/unoplatform/uno/files/15258165/net8.0-windows10.0.22621-Debug-win-arm64.zip)

attached is a binlog.  Here is my csproj for reference.

```
<Project Sdk="Uno.Sdk">
	<PropertyGroup>
		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows')) or '$(EnableWindowsTargeting)' == 'true'">net8.0-windows10.0.22621</TargetFrameworks>
		<TargetFrameworks>
			net8.0-android;
			net8.0-ios;
			net8.0-maccatalyst;
			net8.0; <!-- needed for testing platform -->
			$(TargetFrameworks);
		</TargetFrameworks>

		<OutputType>Exe</OutputType>
		<UnoSingleProject>true</UnoSingleProject>

		<!-- Display name -->
		<ApplicationTitle>ZiiDMSApp</ApplicationTitle>
		<!-- App Identifier -->
		<ApplicationId>com.ZiiDMS.ZiiDMSApp</ApplicationId>
		<!-- Versions -->
		<ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
		<ApplicationVersion>1</ApplicationVersion>
		<!--
      If you encounter this error message:

        error NETSDK1148: A referenced assembly was compiled using a newer version of Microsoft.Windows.SDK.NET.dll.
        Please update to a newer .NET SDK in order to reference this assembly.

      This means that the two packages below must be aligned with the "build" version number of
      the "Microsoft.Windows.SDK.BuildTools" package above, and the "revision" version number
      must be the highest found in https://www.nuget.org/packages/Microsoft.Windows.SDK.NET.Ref.
    -->
		<!-- <WindowsSdkPackageVersion>10.0.22621.28</WindowsSdkPackageVersion> -->

		<!--
      UnoFeatures let's you quickly add and manage implicit package references based on the features you want to use.
      https://aka.platform.uno/singleproject-features
    -->
		<UnoFeatures>
			Hosting;
			Toolkit;
			Logging;
			MauiEmbedding;
			Mvvm;
			Configuration;
			Localization;
			ExtensionsCore;
		</UnoFeatures>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)'=='Debug' or '$(IsUiAutomationMappingEnabled)'=='True'">
		<IsUiAutomationMappingEnabled>True</IsUiAutomationMappingEnabled>
		<DefineConstants>$(DefineConstants);USE_UITESTS</DefineConstants>
	</PropertyGroup>

	<ItemGroup>

		<PackageReference Include="Microsoft.Extensions.Logging.Console" />
		<PackageReference Include="Microsoft.Graph" />

		<PackageReference Include="Microsoft.Extensions.DependencyInjection" />

		<PackageReference Include="Microsoft.Extensions.Http" />
		<PackageReference Include="Microsoft.Extensions.Http.Resilience" />

		<!-- Community Toolkit stuff -->
		<PackageReference Include="CommunityToolkit.Mvvm" />
		<PackageReference Include="CommunityToolkit.WinUI.Animations" />
		<PackageReference Include="CommunityToolkit.WinUI.Controls.SettingsControls" />
	</ItemGroup>

	<!-- this is for windows specific packages -->
	<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
		<!-- WinUICommunity packages -->
		<PackageReference Include="WinUICommunity.Core" />
		<PackageReference Include="WinUICommunity.Components" />

	</ItemGroup>

	<ItemGroup>
		<ProjectReference Include="..\..\..\..\Framework\Client\ZtClientFramework.Maui\ZtClientFramework.Maui.csproj" />
		<ProjectReference Include="..\..\..\..\Framework\Client\ZtClientFramework\ZtClientFramework.csproj" />
		<ProjectReference Include="..\..\..\..\Framework\Shared\ZtFramework.BRPInterfaces\ZtFramework.BRPInterfaces.csproj" />
		<ProjectReference Include="..\..\..\..\Framework\Shared\ZtFramework.Settings\ZtFramework.Settings.csproj" />
		<ProjectReference Include="..\..\..\..\Framework\Shared\ZtFramework\ZtFramework.csproj" />
		<ProjectReference Include="..\..\..\..\GlobalResources\GlobalResources.csproj" />
		<ProjectReference Include="..\..\..\..\ToBeDeleted\AppShare\AppShare.csproj" />
		<ProjectReference Include="..\..\..\Common\ZiiDMSApp.Common\ZiiDMSApp.Common.csproj" />
	</ItemGroup>

</Project>
```

**Comment by jeromelaban** [comment]
Thanks for the report. Could you provide the command line you're using to build?

**Reply by cconner100** [reply]
I am just hitting debug from vs2022, the output is because i set the logging to diagnostic, if i need to do something else let me know

**Reply by cconner100** [reply]
OK here is an additional buildlog using this command line, the error still occurs

MSBuild.exe /restore /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading  /p:Configuration=Release;Platform="Any CPU" /p:AppxPackageDir=C:\temp\Output\ /p:AppxPackageSigningEnabled=false /r /bl
[msbuild.zip](https://github.com/unoplatform/uno/files/15269461/msbuild.zip)

**Comment by cconner100** [comment]
I tried a new uno project from scratch, checked my system with Uno-check, then created a new project with the vs2022 extension.  It gets the same error.  I also installed vs2022 released and vs2022 pre-release both get the same issue. attached is the standard uno solution.

[ControlTestApp.zip](https://github.com/unoplatform/uno/files/15270886/ControlTestApp.zip)

**Reply by jeromelaban** [reply]
Thanks for the update. This was a good opportunity to create docs based on this feedback.

Can you try what's listed in this PR: https://github.com/unoplatform/uno/pull/16678

Thanks!

**Reply by cconner100** [reply]
Looking at the PR it states to open the Package.appmanifest, did that but still get the same error, not sure if this is the same issue.  For some reason i have projects that work correctly only these 2 new ones that i created have an issue, these 2 new ones with the issue were created with the newest package available.

Not sure what else you want me to do to resolve this issue.

**Reply by jeromelaban** [reply]
Thanks. Can you provide a binlog again with the commands specified in the docs PR?

**Reply by cconner100** [reply]
Thanks for the response but i think we are talking about different issues, first trying to follow the docs on what your pointing to on the pr.  If i do the following, Create a new Uno Project from scratch named CantDeploy and run the command in the pr

msbuild /r /p:TargetFramework=net8.0-windows10.0.19041 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms="x86|x64|arm64" /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:\temp\output\\" /p:AppxPackageSigningEnabled=true

I get the following error
"C:\Users\curtisconner\source\repos\CantDeploy\CantDeploy.sln" (Restore target) (1) ->
(ValidateSolutionConfiguration target) ->
  C:\Users\curtisconner\source\repos\CantDeploy\CantDeploy.sln.metaproj : error MSB4126: The specified solution configuration
"Release|x64" is invalid. Please specify a valid solution configuration using the Configuration and Platform properties (e.g.
MSBuild.exe Solution.sln /p:Configuration=Debug /p:Platform="Any CPU") or leave those properties blank to use the default solu
tion configuration. [C:\Users\curtisconner\source\repos\CantDeploy\CantDeploy.sln]

which makes sense since your project creation only has Any CPU defined in the solution configuration.


Next the error is not that i cant build, i can build i just cant deploy, when i build i get a clean build.  But when i try to deploy either from VS2022 with build|deploy or from trying just to run the debugger i get the following error

DEP1700: The recipe file "C:\Users\curtisconner\source\repos\CantDeploy\CantDeploy\bin\Debug\net8.0-windows10.0.19041\CantDeploy.build.appxrecipe" does not exist. You may need to build your project.

Again, in 2 other projects this works just fine, its just the newest projects i created using the newest configuration brought down with Uno-check.

what else can i send you since the command in the PR is not valid for the standard solution generated by uno.

If i am doing something wrong let me know

**Reply by jeromelaban** [reply]
Thanks. The PR command is missing something where `/p:AppxPackageDir="C:\temp\output\"` should be `/p:AppxPackageDir="C:/temp/output/"` to avoid `makeappx.exe` errors, but aside from that, I'm not getting the errors you're getting.

Note that the solution does not need to have an x64 Platform defined for the msbuild command to work on the command line.

How are you creating the project? If you're using `dotnet new` can you do a `dotnet new update` to make sure you're the latest? Otherwise, make sure to run the same command that's failing with `/bl` and [post the binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html), it will help troubleshoot further.

**Reply by cconner100** [reply]
ok so this is strange.

1. VS2022 of my project builds without any warnings or errors
2. Deploying the project from VS2022 fails with above error
3. Running the command line msbuild /r /p:TargetFramework=net8.0-windows10.0.22621 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms="x86|x64|arm64" /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=true /bl 
Fails with the following error
"C:\Users\curtisconner\source\repos\ZiiDMSV2\Framework\Shared\ZtFramework\ZtFramework.csproj" (default target) (5:22) ->
(ResolvePackageAssets target) ->
  C:\Program Files\dotnet\sdk\8.0.204\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(266,5): error NETSDK1005: Assets file 'C:\Users\curtisconner\source\rep
os\ZiiDMSV2\Framework\Shared\ZtFramework\obj\project.assets.json' doesn't have a target for 'net8.0'. Ensure that restore has run and that you have included 'net8.0' in the TargetFramewo
rks for your project. [C:\Users\curtisconner\source\repos\ZiiDMSV2\Framework\Shared\ZtFramework\ZtFramework.csproj::TargetFramework=net8.0]

Yet my project file for ZtFramework is simple

```
<PropertyGroup>
	<TargetFrameworks>netstandard2.0;net8.0</TargetFrameworks>
	<Configurations>Debug;Release</Configurations>
	<EnforceCodeStyleInBuild>True</EnforceCodeStyleInBuild>
	<EnableNETAnalyzers>True</EnableNETAnalyzers>
	<AnalysisLevel>latest</AnalysisLevel>
	<AssemblyName>ZtFramework</AssemblyName>
	<RootNamespace>ZtFramework</RootNamespace>
	<LangVersion>latest</LangVersion>
	<Nullable>enable</Nullable>
	<!--<GeneratePackageOnBuild>True</GeneratePackageOnBuild>
	<Title>ZiiDMS Base Framework</Title>
	<Company>Ziios Thailand</Company>
	<Description>Base framwwork for ZiiDMS</Description>
	<Copyright>2023 Ziios Thailand</Copyright>
	<IncludeSymbols>True</IncludeSymbols>
	<Version>1.0.1</Version>
	--><!--<PackageIcon>ZiiosFullLogo81x32.png</PackageIcon>--><!--
	<Authors>Ziios Thailand</Authors>
	<IsPackable>true</IsPackable>-->
	<ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
</PropertyGroup>
```

**Reply by cconner100** [reply]
here is the binlog
[msbuild.zip](https://github.com/unoplatform/uno/files/15289461/msbuild.zip)

**Reply by cconner100** [reply]
OK built a new vm, install from scratch vs2022 preview, installed uno-check, installed the wizzard, created a new project and i get the same error, completely baffled. FYI i am running on Parallels with windows arm64 and vs arm64. 

frustrated

**Reply by jeromelaban** [reply]
Thanks for the binlog. It seems indeed that there are two issues. The first where it does not build under visual studio is something I'm not able to repro, but the second issue (`doesn't have a target for 'net8.0'`) is related to the `-p:TargetFramework` parameter and is a derivative of this WinAppSDK issue: https://github.com/microsoft/WindowsAppSDK/issues/3548, likely happening because you have separate `net8.0`-only projects.

Let's try something:
- Build first with the following command, to restore nuget packages:
  ```
  msbuild /p:TargetFramework=net8.0-windows10.0.22621 /p:Configuration=Release /p:Platform=x64 /t:Restore
  ```
- Then run the following command:
  ```
  msbuild /p:TargetFramework=net8.0-windows10.0.22621 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms="x86|x64|arm64" /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=true /bl
  ```
  Note that this command does not run with `/r`, like previously suggested.

**Reply by cconner100** [reply]
OK did the outlines steps in your mail, but wanted to update you on one issue, it CAN build in vs2022 but can not deploy.

attached is the binlog, but it still gets the .net8 error
[msbuild.zip](https://github.com/unoplatform/uno/files/15300999/msbuild.zip)


The error seems to be more of this https://github.com/microsoft/microsoft-ui-xaml/issues/7754, again it can build just fine and i can build and run on ios, android, maccatalyst.  Its something to do with the packaging.  digging in more to this now

**Reply by cconner100** [reply]
ok after messing with it i got the following cmd to run

msbuild /p:TargetFramework=net8.0-windows10.0.22621 /p:Configuration=Release /p:Platform=x64 /p:GenerateAppxPackageOnBuild=true /p:AppxBundlePlatforms="x86|x64|arm64" /p:AppxBundle=Never /p:UapAppxPackageBuildMode=Sideloading /p:AppxPackageDir="C:/temp/output/" /p:AppxPackageSigningEnabled=true /bl

it compiles with no errors, attached is the binlog
[msbuild.zip](https://github.com/unoplatform/uno/files/15303307/msbuild.zip)

but in the output directory i still dont see the .build.appxrecipe and vs2022 wont do a deploy because of this issue

if i run the command 
msbuild /t:Rebuild /p:Configuration=Debug;Platform="net8.0"

I get the following error

C:\Users\curtisconner\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.WindowsAppSDK.SelfCont
ained.targets(212,9): error MSB4018: The "GenerateAppManifestFromAppx" task failed unexpectedly. [C:\Users\curtisconner\source
\repos\ZiiDMSV2\Client\Mobile\ZiiDMSPhoneApp\ZiiDMSPhoneApp\ZiiDMSPhoneApp.csproj::TargetFramework=net8.0-windows10.0.22621]
C:\Users\curtisconner\.nuget\packages\microsoft.windowsappsdk\1.5.240404000\buildTransitive\Microsoft.WindowsAppSDK.SelfContai
ned.targets(212,9): error MSB4018: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\curtisconner\source\repos\ZiiDMSV2\Client\Mobile\ZiiDMSPhoneApp\ZiiDMSPhoneApp\obj\net8.0\Debug\net8.0-Windows10.0.22621\MsixContent\AppxManifest.xml'.

which leads me to the conclusion that the project is building correctly but for some reason its not processing the AppManifest correctly.

Looking in the build log i see this
9>Target Build:
9>Target "_PackAsBuildAfterTarget" skipped, due to false condition; ('$(GeneratePackageOnBuild)' == 'true' AND '$(IsInnerBuild)' != 'true') was evaluated as ('false' == 'true' AND '' != 'true').
9>Target Rebuild:

It seems to be skipping the packaging for some reason,

attached is the build output log
[Output-Build.zip](https://github.com/unoplatform/uno/files/15303359/Output-Build.zip)

**Reply by cconner100** [reply]
One more piece of information, i created a standard C# WinUI3 app using the vs2022 wizard, it creates the package and runs so i now know my environment works for package deployment, seems to be something with the uno project.

In addition in the normal WinUI3 project i see
Property reassignment: $(PackageAction)="_GenerateAppxPackage" (previous value: "Pack") at C:\Users\curtisconner\.nuget\packages\microsoft.windowsappsdk\1.5.240428000\buildTransitive\Microsoft.Build.Msix.Packaging.targets (117,5)

which there is no reference to this in the Uno project, what fires off the GenerationAppxPackage in Uno? In WinUI3 app this is where it generates the apprecipe file which i am missing

**Reply by cconner100** [reply]
More information, I just installed a fresh windows 11 intel machine, installed VS2022 enterprise release, installed uno check and made sure uno check ran clean.  Created a new project using the VS2022 wizard and get the same error.  I can run on android with this project but windows complains about appxrecipe is not found.  Unpackaged runs fine.

**Reply by jeromelaban** [reply]
Thanks. The following command `msbuild /t:Rebuild /p:Configuration=Debug;Platform="net8.0"` is not valid, `Platform` must be either `x64`, `x86` or `arm64`.

> attached is the build output log 
[Output-Build.zip](https://github.com/unoplatform/uno/files/15303359/Output-Build.zip)attached is the build output log

Could you provide the binlog instead? This particular log is missing information, thanks!

**Reply by jeromelaban** [reply]
Additionally, I'm still not able to reproduce any of what you're reporting and about :

> More information, I just installed a fresh windows 11 intel machine, installed VS2022 enterprise release, installed uno check and made sure uno check ran clean. Created a new project using the VS2022 wizard and get the same error. I can run on android with this project but windows complains about appxrecipe is not found. Unpackaged runs fine.

Could you make a video of what you're doing, there's definitely something that's different that we're missing.

**Reply by cconner100** [reply]
OK here ya go

Video on creating new Uno project and trying to run
[https://packingslips.blob.core.windows.net/video/PackageBuild.mp4](url)

Video trying to use the above command lines that you gave me
[https://packingslips.blob.core.windows.net/video/CommandLine.mp4](url)

Note: copy and paste the above links into a browser, not sure why github is blocking

binlog of the last run on the video
[msbuild.zip](https://github.com/unoplatform/uno/files/15315773/msbuild.zip)

Note all the above was run on an Asus 15 laptop I7 and 16gb memory.  Windows 11 with all service packs applied, Visual Studio 2022 Enterprise with all service packs applied, uno-check run clean and only using the code created by the uno wizard.

Let me know what else i can supply.

**Reply by jeromelaban** [reply]
Thanks, this helps.

For the command line part, make sure to run the `msbuild` command in the folder of the app head which contains the `.csproj`, not in the solution folder. Otherwise, `msbuild` will try to find a solution configuration that matches the `Configuration` parameter, which is not present. It's written in the doc that is in the PR, but it's likely not visible enough, we'll make changes.

I can confirm that the appxrecipe error, we'll be looking into it.

**Reply by jeromelaban** [reply]
The `appxrecipe` error is related specificly to enabling MAUI embedding. The workaround is to add the following in the app's `csproj`:

```xml
<PropertyGroup>
   <EnableMsixTooling>true</EnableMsixTooling>
</PropertyGroup>
```

We'll be fixing this in https://github.com/unoplatform/uno.extensions/pull/2295.

**Reply by jeromelaban** [reply]
The extensions fix is now published. You can update your `<ExtensionsVersion>` in `Directory.Build.props` to `4.1.23` to fix the MAUI embedding build issue.

**Reply by cconner100** [reply]
Thanks, back working again.
